dc-commons-xml/src/main/java/de/digitalcollections/commons/xml/xpath/ReflectionUtils.java

```diff
@@ -0,0 +1,81 @@
+package de.digitalcollections.commons.xml.xpath;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Some helper methods to deal with reflection
+ */
+public class ReflectionUtils {
+
+  /**
+   * @param type The type to be evaluated
+   * @param annotation the annotation class
+   * @return a list of all methods, which carry the given annotation
+   */
+  public static List<Method> getMethodsAnnotatedWith(final Class<?> type, final Class<? extends Annotation> annotation) {
+    final List<Method> methods = new ArrayList<Method>();
+    Class<?> klass = type;
+    while (klass != Object.class) { // need to iterated thought hierarchy in order to retrieve methods from above the current instance
+      // iterate though the list of methods declared in the class represented by klass variable, and add those annotated with the specified annotation
+      final List<Method> allMethods = new ArrayList<Method>(Arrays.asList(klass.getDeclaredMethods()));
+      for (final Method method : allMethods) {
+        if (method.isAnnotationPresent(annotation)) {
```

Comment:
Würrde hier ne guard clause mit continue machen, spart ne Verschachtelung.

Oder  alternativ eleganter mit Streaming API:

```java
Arrays.stream(klass.getDeclaredMethods())
    .filter(m -> m.isAnnotationPresent(annotation))
    .forEach(m -> {
      Annotation anno = m.getAnnotation(annotation);
      // TODO: Process anno
      methods.add(m);
    })
```

Reply:
Du müsstest hier auch noch checken dass die Methode ein Setter ist der eines der supporteten Argumente ist, ist das mit `TODO: Process acno` gemeint? Das könnte man super in eine Prädikat-Methode packen, was auch für die Streaming API sprechen würde.

Reply:
Umbau auf StreamingAPI ist erfolgt; Umbau auf Validierung kommt später